src/util/mutationUpdate.ts

```diff
@@ -0,0 +1,37 @@
+import { InfiniteData } from '@tanstack/react-query'
+import { SearchData } from 'src/types/SearchData'
+import { Page } from 'src/types/SearchMeta'
+
+/**
+ * Updates a single item in the search data.
+ * @param newItem - The new item to update.
+ * @returns A function that takes the previous search data and returns the updated search data with the new item.
+ */
+export function updateQueryItem<T extends { id: number }>(newItem: T) {
+  return (prev: SearchData<T>) => {
+    return {
+      ...prev,
```

Comment:
do you have a link or mind explaining what this "..."syntax is or does?

Reply:
It's called the spread operator. Here I'm using it to create a new object with all the properties of `prev`, followed by overridden values.

Here's an article talking about the many uses of the spread operator.

https://kinsta.com/knowledgebase/spread-operator-javascript/

https://kinsta.com/knowledgebase/spread-operator-javascript/#3-merging-and-cloning-objects